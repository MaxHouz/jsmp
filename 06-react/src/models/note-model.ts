export class NoteModel {
    constructor(
        public title: string,
        public text: string,
        public date: any,
        public done: boolean,
        public archieved: boolean,
        public id?: number
    ){}
}