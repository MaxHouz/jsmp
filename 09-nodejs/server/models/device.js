const mongoose = require('mongoose');

const Device = mongoose.model('Device', {
    name: String,
    address: String,
    port: Number,
    state: String,
    groupId: String,
    log: [{
        date: String,
        action: String
    }]
});

module.exports = Device;
