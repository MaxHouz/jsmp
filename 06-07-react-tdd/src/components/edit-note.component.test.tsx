import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { EditNoteCompenent, mapDispatchToProps } from './edit-note.component';
import { NoteModel } from '../models/note-model';

configure({adapter: new Adapter()});
describe('EditNoteCompenent', () => {
    let component: any;
    const editNoteMock = jest.fn();
    const routerMock = {
        params: {
            id: 4
        }
    }
    const mockEvent = {
        target: {
            value: 'mockValue'
        }
    }
    const historyMock = {
        push: jest.fn()
    }

    beforeEach(() => {
        component = shallow(
            <EditNoteCompenent
                editNote={editNoteMock}
                // @ts-ignore
                match={routerMock}
                // @ts-ignore
                history={historyMock}
            />
        )
    })

    it('should render component', () => {
        expect(component).toMatchSnapshot();
    })

    it('should update title value on change', () => {
        component.find('#note-form-title').simulate('change', mockEvent);
        component.update();

        expect(component.state().titleValue).toBe(mockEvent.target.value);
    })

    it('should update text value on change', () => {
        component.find('#note-form-text').simulate('change', mockEvent);
        component.update();

        expect(component.state().textValue).toBe(mockEvent.target.value);
    })

    it('should edit note', () => {
        component.state().titleValue = 'mockTitleValue';
        component.state().textValue = 'mockTextValue';
        component.state().note = {
            done: true,
            archieved: true,
            id: 3
        }

        component.find('Button').at(0).simulate('click');
        const newNoteMock = {
            title: 'mockTitleValue',
            text: 'mockTextValue',
            done: true,
            archieved: true,
            id: 3
        }

        expect(editNoteMock).toHaveBeenCalledWith(newNoteMock);
    })

    it('should navigate to home page after note edit', () => {
        component.state().note = {
            done: true,
            archieved: true,
            id: 3
        }

        component.find('Button').at(0).simulate('click');
        expect(historyMock.push).toHaveBeenCalledWith('/');
    })

    it('should map dispatch to props', () => {
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).editNote({} as NoteModel);
        
        expect(typeof dispatch.mock.calls[0][0]).toBe('function');
    })
    
})