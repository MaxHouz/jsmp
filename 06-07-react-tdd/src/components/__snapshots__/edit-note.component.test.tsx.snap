// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditNoteCompenent should render component 1`] = `ShallowWrapper {}`;
